### PR TITLE
Add and consolidate missing multimedia logging

### PR DESCRIPTION
**Pull request by orangejenny**
Realized https://github.com/dimagi/commcare-hq/pull/23287/ isn't notifying the dev team when a CCZ is built incorrectly.

Also added a tag to all of the emails related to this issue, as it's getting difficult for me to manage searching my inbox for them. I don't intend this to be permanent - some of these emails can be removed altogether once this bug is sorted out, others can just have the tag removed.

@mkangia / @snopoke 
code buddy @millerdev

### CODE REVIEW

File: corehq/apps/linked_domain/remote_accessors.py [hunk 1]
```diff
@@ -58,18 +59,30 @@ def pull_missing_multimedia_for_app(app):
...
+                False, "[ICDS-291] Multimedia pull successful", json.dumps([{
+                    'domain': app.domain,
+                    'app_id': app.get_id,
+                    'version': app.version,
+                    'length of multimedia_map': len(app.multimedia_map),
+                    'number of missing files pulled': len(missing_media),
+                }, {
+                    'missing': missing_media,
+                }], indent=4)
+            )
```
Comment (mkangia) [inline corehq/apps/linked_domain/remote_accessors.py]: why do we need this? notifying a successful pull?

**Review by mkangia — APPROVED** [review]
nit: i don't think `[ICDS-291]` is the right thing to add in all subjects/messages. It's possible we would start either including errors from other domains to this or being made to look at errors from other domains as well thinking its from icds. May be check for domain and then append?

**Comment by orangejenny** [comment]
> i don't think `[ICDS-291]` is the right thing to add in all subjects/messages.

This might be laziness talking, but I'm inclined to leave it as-is. I started by using "missing multimedia" as the tag but decided something more unique would be better - I literally want to be able to search my email and get everything related to this bug and nothing else. I'd rather not add a bunch of domain checks to the code, it's already cluttered with the new logging.

File: corehq/apps/linked_domain/remote_accessors.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/linked_domain/remote_accessors.py]: One of my theories is that the list of missing files is sometimes incorrect or incomplete, so the pull seems "successful" but didn't get all the right files.

Comment (mkangia) [reply corehq/apps/linked_domain/remote_accessors.py]: cool.. sounds right..

File: corehq/apps/hqmedia/tasks.py [hunk 2]
```diff
@@ -183,7 +185,9 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
                 with zipfile.ZipFile(tmp, "r") as z:
                     media_suites = [f for f in z.namelist() if re.search(r'\bmedia_suite.xml\b', f)]
                     if len(media_suites) != 1:
-                        errors.append(_('Could not identify media_suite.xml in CCZ'))
+                        message = _('Could not identify media_suite.xml in CCZ')
+                        errors.append(message)
+                        notify_exception(None, "[ICDS-291] ".format(message))
```
Comment (millerdev) [inline corehq/apps/hqmedia/tasks.py]: `"[ICDS-291] ".format(message)` will always result in `"[ICDS-291] "`.

Did you want `"[ICDS-291] {}".format(message)`

File: corehq/apps/hqmedia/views.py [hunk 3]
```diff
@@ -734,7 +734,7 @@ def _media_files():
                     'error': e,
                 }
                 errors.append(message)
-                notify_exception(None, message)
+                notify_exception(None, "[ICDS-291] ".format(message))
```
Comment (millerdev) [inline corehq/apps/hqmedia/views.py]: same

File: corehq/apps/hqmedia/views.py [hunk 4]
```diff
@@ -935,7 +935,7 @@ def _download_index_files(app, build_profile_id, is_retry=False):
     try:
         files = _download_index_files(app, build_profile_id)
     except Exception as e:
-        notify_exception(None, e.message)
+        notify_exception(None, "[ICDS-291] ".format(six.text_type(e)))
```
Comment (millerdev) [inline corehq/apps/hqmedia/views.py]: same

File: corehq/apps/hqmedia/tasks.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/hqmedia/tasks.py]: That's embarrassing. Thank you!

**Pull request merged**